sigmastate/src/main/scala/sigmastate/lang/Terms.scala

```diff
@@ -113,6 +117,26 @@ object Terms {
       case _ => NoType
     }
     override def opType: SFunc = SFunc(Vector(func.tpe +: args.map(_.tpe):_*), tpe)
+
+    protected final override def eval(E: ErgoTreeEvaluator, env: DataEnv): Any = {
+      if (args.isEmpty) {
+        // TODO coverage
+        val fV = func.evalTo[() => Any](E, env)
+        fV()
+      }
+      else if (args.length == 1) {
+        val fV = func.evalTo[Any => Any](E, env)
+        val argV = args(0).evalTo[Any](E, env)
+        fV(argV)
+      }
+      else {
+        // TODO coverage
+        val f = func.evalTo[Seq[Any] => Any](E, env)
+        val argsV = args.map(a => a.evalTo[Any](E, env))
+        f(argsV)
+      }
+      // TODO JITC
```

Comment:
Added separate issue https://github.com/ScorexFoundation/sigmastate-interpreter/issues/662 to implement all TODO JITC in a separate PR